lib/matplotlib/hatch.py

```diff
@@ -181,6 +183,20 @@ def __init__(self, hatch, density):
     ]
 
 
+def _validate_hatch_pattern(hatch):
+    valid_hatch_patterns = set(r'-+|/\xXoO.*')
+    if hatch is not None:
+        invalids = set(hatch).difference(valid_hatch_patterns)
+        if invalids:
+            cbook.warn_deprecated(
+                '3.4',
+                message='hatch must consist of a string of '
+                        f'"{"".join(valid_hatch_patterns)}" or None, but '
+                        f'found "{"".join(invalids)}". Passing invalid values '
```

Comment:
```suggestion
                        f'found the following invalid values "{"".join(invalids)}". Passing invalid values '
```
to be clear `invalids` contains only invalid values and not all the passed in values.